src/CommonITILValidation.php

```diff
@@ -1053,61 +1085,88 @@ public function showSummary(CommonDBTM $item)
+            $edit_button = "";
             if ($canedit) {
-                $edit_title = __s('Edit');
-                $item_id = (int) $item->fields['id'];
-                $row_id = (int) $row["id"];
-                $rand = htmlescape($rand);
-                $view_validation_id = htmlescape($this->fields[static::$items_id]);
-                $root_doc = htmlescape($CFG_GLPI["root_doc"]);
-                $params_json = json_encode([
-                    'type'             => static::class,
-                    'parenttype'       => static::$itemtype,
-                    static::$items_id  => $this->fields[static::$items_id],
-                    'id'               => $row["id"],
-                ]);
-
-                $script = <<<HTML
-                    <span class="ti ti-edit" style="cursor:pointer" title="{$edit_title}" 
-                          onclick="viewEditValidation{$item_id}{$row_id}{$rand}();" 
-                          id="viewvalidation{$view_validation_id}{$row_id}{$rand}">
-                    </span>
-                    <script>
-                        function viewEditValidation{$item_id}{$row_id}{$rand}() {
-                            $('#viewvalidation{$item_id}{$rand}').load('$root_doc/ajax/viewsubitem.php', $params_json);
-                        };
-                    </script>
-HTML;
+                $edit_button = $this->getModalFormHtmlElements('editapproval_modal', __('Edit Approval'), $itil->getID(), $validation["id"]) + ['anchor_title' => __('Edit') ];
             }
 
-            $values[] = [
-                'edit'                  => $script,
+            $itil_validationstep_id = $validation['itils_validationsteps_id'];
+            $validations[$itil_validationstep_id]['entries'][] = [
+                'edit'                  => $edit_button,
                 'status'                => $status,
                 'type_name'             => $type_name,
                 'target_name'           => $target_name,
                 'is_answered'           => $is_answered,
                 'comment_submission'    => $comment_submission,
                 'comment_validation'    => $comment_validation,
                 'document'              => $document,
-                'submission_date'       => $row["submission_date"],
-                'validation_date'       => $row["validation_date"],
-                'user'                  => getUserName($row["users_id"]),
+                'submission_date'       => $validation["submission_date"],
+                'validation_date'       => $validation["validation_date"],
+                'user'                  => getUserName($validation["users_id"]),
             ];
+
+            $itil_edit_js_identifier = 'itilvalidation_edit_' . $itil_validationstep_id;
+            $url = $itil::getValidationStepClassName()::getFormURLWithId($itil_validationstep_id);
+            $validations[$itil_validationstep_id]['edit_link_js']['js'] = Ajax::createIframeModalWindow(
```

Comment:
Passing raw JS to Twig like this seems odd. Try using existing Twig templates or call the needed PHP function from Twig.